pageserver/src/tenant/ephemeral_file/page_caching.rs

```diff
@@ -0,0 +1,105 @@
+//! Wrapper around [`super::zero_padded_read_write::RW`] that uses the
+//! [`crate::page_cache`] to serve reads that need to go to the underlying [`VirtualFile`].
+
+use crate::context::RequestContext;
+use crate::page_cache::{self, PAGE_SZ};
+use crate::tenant::block_io::BlockLease;
+use crate::virtual_file::VirtualFile;
+
+use std::io;
+use tracing::*;
+
+use super::zero_padded_read_write;
+
+/// See module-level comment.
+pub struct RW {
+    page_cache_file_id: page_cache::FileId,
+    rw: super::zero_padded_read_write::RW,
+}
+
+impl RW {
+    pub fn new(file: VirtualFile) -> Self {
+        Self {
+            page_cache_file_id: page_cache::next_file_id(),
+            rw: super::zero_padded_read_write::RW::new(file),
+        }
+    }
+
+    pub fn page_cache_file_id(&self) -> page_cache::FileId {
+        self.page_cache_file_id
+    }
+
+    pub(crate) async fn write_all_borrowed(&mut self, srcbuf: &[u8]) -> Result<usize, io::Error> {
+        // It doesn't make sense to proactively fill the page cache on the Pageserver write path
+        // because Compute is unlikely to access recently written data.
+        self.rw.write_all_borrowed(srcbuf).await
+    }
```

Comment:
Then again, this PR is showing a regression over the current `main` baseline.

https://neondb.slack.com/archives/C033RQ5SPDH/p1713890189520339?thread_ts=1713433920.576379&cid=C033RQ5SPDH

=> @ reviewers, please continue your reviews, but let's not merge this until the reason is understood

Reply:
https://github.com/neondatabase/neon/pull/7511

Will merge it into this one.